public/app/features/dashboard/state/PanelModel.ts

```diff
@@ -431,7 +447,7 @@ export class PanelModel implements DataConfigSource, IPanelModel {
     const version = getPluginVersion(plugin);
 
     if (this.autoMigrateFrom) {
-      const wasAngular = this.autoMigrateFrom === 'graph' || this.autoMigrateFrom === 'table-old';
+      const wasAngular = autoMigrateAngular[this.autoMigrateFrom] != null;
       this.callPanelTypeChangeHandler(
```

Comment:
Have we ever auto migrated angular panels to another panel type in the past? If so it may be useful to look at the approach we took and learn from it for this time around

Reply:
graph + table-old have been loading this way for a few years, heatmap for 6+ months

The difference here making the pattern more general and adding worldmap+piechart explicitly